src/components/10-atoms/input-text/index.js

```diff
@@ -248,6 +252,33 @@ class AXAInputText extends AXAPopupMixin(NoShadowDOM) {
     this.nativeInput = this.querySelector('input');
   }
 
+  _formatCurrency(value) {
+    const { currency, type } = this;
+
+    if (currency && type === 'text') {
+      const hasAtLeastOneDigit = /\d/.test(value);
+      const dotsCounted = (value.match(/\./g) || []).length;
+
+      if (!this.currencyFormatter) {
+        // just create a new Intl if it does not exist
+        this.currencyFormatter = new Intl.NumberFormat('de-CH', {
+          style: 'currency',
+          currency,
+        });
+      }
+
+      if (hasAtLeastOneDigit && dotsCounted <= 1) {
+        this.invalid = false;
+        const valueDecimalsOnly = value.replace(/[^0-9.]/g, '');
+        return this.currencyFormatter.format(valueDecimalsOnly);
```

Comment:
This is a brittle way to solve the NaN bug that arises with a specific input that has multiple dots. There could be many other patterns.

You need to address the root of the problem with `parseFloat(valueDecimalsOnly)`. If it  parses as a valid floating-point number, it could be used as a currency value.